ffi/nodejs/swig.i

```diff
@@ -18,4 +18,17 @@
     $result = resolver->GetPromise();
 %}
 
+%inline %{
+    uint8_t *uint_array(int size) {
+        return (uint8_t *) malloc(size*sizeof(uint8_t));
+    }
+    void uint_array_set(uint8_t *a, int i, int val) {
+        a[i] = val;
+    }
```

Comment:
This is very inefficient way of copying stuff; we need to use `memcpy` instead to copy the whole piece

Reply:
I can switch from the direct assignment to using `memcpy` like this:
```
void uint_array_set(uint8_t *a, int i, int val) {
    memcpy ( &a[i], &val, sizeof(uint8_t) );
}
```
but I don't think we can use `memcpy` to copy the whole piece. in node.js pointers do not exist, this is the reason I had to copy value by value, because conversion of js integers to C++ integers is already handled by swig, while array conversion is not handled by swig

Reply:
That is really strange that with native modules for NodeJS you can't access array pointers.

Reply:
@dr-orlovsky I've already spent two days on this, so for now, if you agree, I'd go with this solution (which is not that inefficient after all)

For future reference:
- on [swig doc](http://www.swig.org/Doc4.0/SWIG.html#SWIG_nn26) they suggest to construct arrays using helper methods (as I did)
- I've tried using [arrays_javascript.i](https://github.com/swig/swig/blob/master/Lib/javascript/v8/arrays_javascript.i), but it's not working as expected (may be bugged)
- I've tried using native modules (`ref-napi` and `ref-array-di`) as shown [here](http://jakegoulding.com/rust-ffi-omnibus/slice_arguments/), but then realized that this way of binding is incompatible with ours (which uses swig)

And please let me know if it still makes sense switching from `a[i] = val;` to `memcpy ( &a[i], &val, sizeof(uint8_t) );`

Reply:
Thank you for the investigation. Ok, let's stick with what we have here. And no, using memcpy for a single-byte copy does not make any sense, so no need for it.